tools/src/test/java/org/apache/kafka/tools/TopicCommandIntegrationTest.java

```diff
@@ -86,652 +90,792 @@
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testDescribeWhenTopicDoesntExistWithIfExists(String quorum) throws ExecutionException, InterruptedException {
-        topicService.describeTopic(buildTopicCommandOptionsWithBootstrap("--describe", "--topic", testTopicName, "--if-exists"));
+    @ClusterTemplate("generate1")
+    public void testDescribeWhenTopicDoesntExistWithIfExists(TestInfo testInfo) throws Exception {
+        String testTopicName = testInfo.getTestMethod().get().getName() + "-" +
+                TestUtils.randomString(10);
+        try (Admin adminClient = clusterInstance.createAdminClient()) {
+            TopicCommand.TopicService topicService = new TopicCommand.TopicService(adminClient);
+
+            topicService.describeTopic(buildTopicCommandOptionsWithBootstrap("--describe", "--topic", testTopicName, "--if-exists"));
+
+            adminClient.close();
+            topicService.close();
+        }
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testDescribeUnavailablePartitions(String quorum) throws ExecutionException, InterruptedException {
-        int partitions = 6;
-        short replicationFactor = 1;
-        TestUtils.createTopicWithAdmin(adminClient, testTopicName, scalaBrokers, scalaControllers, partitions, replicationFactor,
-                scala.collection.immutable.Map$.MODULE$.empty(), new Properties()
-        );
-        try {
+    @ClusterTemplate("generate1")
+    public void testDescribeUnavailablePartitions(TestInfo testInfo) throws ExecutionException, InterruptedException {
+        String testTopicName = testInfo.getTestMethod().get().getName() + "-" +
+                TestUtils.randomString(10);
+
+        try (Admin adminClient = clusterInstance.createAdminClient()) {
+            int partitions = 6;
+            short replicationFactor = 1;
+
+            adminClient.createTopics(Collections.singletonList(new NewTopic(testTopicName, partitions, replicationFactor)));
+            clusterInstance.waitForTopic(testTopicName, partitions);
+
             // check which partition is on broker 0 which we'll kill
             TopicDescription testTopicDescription = adminClient.describeTopics(Collections.singletonList(testTopicName))
-                .allTopicNames().get().get(testTopicName);
+                    .allTopicNames().get().get(testTopicName);
             int partitionOnBroker0 = testTopicDescription.partitions().stream()
-                .filter(partition -> partition.leader().id() == 0)
-                .findFirst().get().partition();
+                    .filter(partition -> partition.leader().id() == 0)
+                    .findFirst().get().partition();
 
-            killBroker(0);
+            clusterInstance.shutdownBroker(0);
+            assertEquals(5, clusterInstance.aliveBrokers().size());
 
             // wait until the topic metadata for the test topic is propagated to each alive broker
```

Comment:
It seems this is equal to `ClusterInstance#waitForTopic`, since it checks the metadata cache for the specific topic/partition

Reply:
Agree, update. Thanks.